Sources/Shared/App.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  App.swift
+//  VimeoNetworking
+//
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+/// The app from which an item, like a video, was originally uploaded, i.e.: iMovie, Vimeo for macOS, Cameo, etc.
+@objc public class App: VIMModelObject {
```

Comment:
Should you just do:
```suggestion
@objcMembers 
public class App: VIMModelObject {
```

Reply:
I also think we should use a more specific name than just `App` since we have several concepts of app in the codebase (at the very least, there's the Vimeo app itself, source app, connected social apps...). Maybe `UploadSourceApp`?

Reply:
Actually, this is more than just the upload source really. What this resource really represents is the app registered with our API. So something more generic than `UploadSourceApp` would be needed. Maybe App _is_ okay?

Reply:
`APIClientApp`, `VimeoClientApp`?

Reply:
`OriginApp`?

Reply:
My vote is for `APIClientApp` or `VimeoClientApp`, I thought of `APIApp` but didn't seem right. `APIClientApp` I think captures what I was thinking of better.

Reply:
Thanks for the discussion on this, I ended up going with `APIClientApp`. I feel like that captures it pretty well.